frontend/packages/console-dynamic-plugin-sdk/src/api/types/statuses.ts

```diff
@@ -0,0 +1,28 @@
+export type StatusComponentProps = {
+  title?: string;
+  iconOnly?: boolean;
+  noTooltip?: boolean;
+  className?: string;
+  popoverTitle?: string;
+};
+
+export type StatusProps = StatusComponentProps & {
+  status: string;
+};
+
+export type StatusIconAndTextProps = StatusComponentProps & {
+  icon?: React.ReactElement;
+  spin?: boolean;
+};
+
+export type GenericStatusProps = StatusComponentProps & {
+  Icon: React.ComponentType<{ title?: string }>;
+  popoverTitle?: string;
+  noTooltip?: boolean;
+};
+
+export type ColoredIconProps = {
+  className?: string;
+  title?: string;
+  size?: 'sm' | 'md' | 'lg' | 'xl';
+};
```

Comment:
We don't typically expose the prop types as API. The dev can use `React.ComponentProps` on the component type to retrieve its prop types.

Reply:
my mistake, I thought this was still in `api.ts` when I was reviewing. Since these aren't exposed as API then it's fine.